lib/maths/CQDigest.cc

```diff
@@ -73,9 +73,15 @@ bool CQDigest::acceptRestoreTraverser(core::CStateRestoreTraverser& traverser) {
         }
     } while (traverser.next());
 
+    this->checkRestoredInvariants();
+
     return true;
 }
 
+void CQDigest::checkRestoredInvariants() const {
+    VIOLATES_INVARIANT_NO_EVALUATION(m_Root, ==, nullptr);
+}
```

Comment:
Maybe this can also call the pre-existing `this->checkInvariants();` and abort if it returns false.